src/Microsoft.AspNet.Mvc.Core/Formatters/StringOutputFormatter.cs

```diff
@@ -31,13 +31,9 @@ public override bool CanWriteResult(OutputFormatterCanWriteContext context)
 
             // Ignore the passed in content type, if the object is string
             // always return it as a text/plain format.
-            if (context.ObjectType == typeof(string))
-            {
-                return true;
-            }
-
-            if (context.Object is string)
+            if (context.ObjectType == typeof(string) || context.Object is string)
```

Comment:
Curious: When is the RHS of the `||` needed? `ObjectType` can't be `object` when `Object` is non-`null`, can it?